// Copyright 2020 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
////////////////////////////////////////////////////////////////////////////////

syntax = "proto3";

package google.crypto.tink;

import "tink/proto/common.proto";

option java_package = "com.google.crypto.tink.proto";
option java_multiple_files = true;
option go_package = "github.com/google/tink/proto/hmac_prf_go_proto";

message HmacPrfParams {
  HashType hash = 1;  // HashType is an enum.
}

// key_type: type.googleapis.com/google.crypto.tink.HmacPrfKey
message HmacPrfKey {
  uint32 version = 1;
  HmacPrfParams params = 2;
  bytes key_value = 3;
}

message HmacPrfKeyFormat {
  HmacPrfParams params = 1;
  uint32 key_size = 2;
  uint32 version = 3;
}
